server/data/data-cassandra/src/main/java/org/apache/james/sieve/cassandra/CassandraSieveRepository.java

```diff
@@ -0,0 +1,276 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.sieve.cassandra;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.backends.cassandra.utils.CassandraConstants;
+import org.apache.james.sieve.cassandra.tables.CassandraSieveTable;
+import org.apache.james.sieverepository.api.ScriptSummary;
+import org.apache.james.sieverepository.api.SieveRepository;
+import org.apache.james.sieverepository.api.exception.DuplicateException;
+import org.apache.james.sieverepository.api.exception.IsActiveException;
+import org.apache.james.sieverepository.api.exception.QuotaExceededException;
+import org.apache.james.sieverepository.api.exception.QuotaNotFoundException;
+import org.apache.james.sieverepository.api.exception.ScriptNotFoundException;
+import org.apache.james.sieverepository.api.exception.StorageException;
+import org.joda.time.DateTime;
+
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+
+public class CassandraSieveRepository implements SieveRepository {
+
+    private final CassandraSieveDAO cassandraSieveDAO;
+
+    @Inject
+    public CassandraSieveRepository(Session session) {
+        this.cassandraSieveDAO = new CassandraSieveDAO(session);
+    }
+
+    @Override
+    public DateTime getActivationDateForActiveScript(String user) throws StorageException, ScriptNotFoundException {
+        return new DateTime(cassandraSieveDAO.getActiveScriptMetadataRow(user)
+            .join()
+            .orElseThrow(ScriptNotFoundException::new)
+            .getDate(CassandraSieveTable.DATE)
+            .getTime());
+    }
+
+    @Override
+    public void haveSpace(String user, String name, long newSize) throws QuotaExceededException, StorageException {
+        throwOnOverQuota(user, spaceThatWillBeUsedByNewScript(user, name, newSize));
```

Comment:
I'd love to do that...
Unfortunately, you will not be able to do that because of  thenApply(throwOnOverQuota) => Exceptions in lambas ;-)